File: acoustics_toolbox_r2020/Python/Bellhop/Tests/DW/Dickins/DickinsBray.py
#=======================================================================
# 
# Bellhop: Dickins Seamount
# Mexilhoeira Grande, Sex Jul 28 17:15:48 WEST 2017
# Written by Tordar 
# 
#=======================================================================

from os import system
from numpy import *
from scipy.io import *
from pylab import *
sys.path.append ("/home/orodrig/FORdoc/at/Python/")
from plotray import *

print("Bellhop - Dickins Seamount:") 
print("ray calculation including bathymetry") 

fid = open('DickinsBray.bty','r')
theline = fid.readline()
theline = fid.readline()
n       = int( theline )
rbtykm  = zeros( n ) 
zbty    = zeros( n )
for i in range(n):
    theline = str( fid.readline() )
    datai = theline.split()
    rbtykm[ i] = float( datai[0] )
    zbty[   i] = float( datai[1] )
fid.close()

rbty = rbtykm*1000

system("bellhop.exe DickinsBray")

figure(1)
plotray('DickinsBray.ray')
plot(rbty,-zbty,'k',linewidth=2)
xlabel('Range (m)')
ylabel('Depth (m)')
title("Bellhop - Dickins Seamount")

show()

print("done.")

File: acoustics_toolbox_r2020/Python/Bellhop/Tests/DW/Dickins/DickinsBray.m
clear all, close all

addpath /home/orodrig/FORdoc/at/Matlab/Plot
addpath /home/orodrig/FORdoc/at/Matlab/ReadWrite

disp('Bellhop - Dickins Seamount:') 
disp('Ray trace run') 
disp('Geometric hat beams in Cartesian coordinates') 

fid = fopen('DickinsBray.bty','r');
s = fscanf(fid,'%s\n',[1 1]);
n = fscanf(fid,'%d\n',[1 1]);
data = fscanf(fid,'%f %f\n',[2 n]);
fclose(fid);

rbtykm = data(1,:); rbty = rbtykm*1000; 
zbty   = data(2,:);

system('bellhop.exe DickinsBray');

plotray('DickinsBray')
hold on
plot(rbtykm,zbty,'k','LineWidth',2)
hold off

disp('done.')
